### PR TITLE
Fix RequestIssue order-of-creation ITF EP code bug

### PR DESCRIPTION
**Pull request by nanotone**
### Description
The new EP code 040SCRGTY requires checking the decision dates on all the RequestIssues on a SupplementalClaim. This PR fixes a bug where `ClaimReview#create_issues!` was saving each RequestIssue and its EPE in lockstep, rather than all RequestIssues followed by all EPEs, resulting in the wrong EP code when the first RequestIssue's EP code depends on the second.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
Added a new test. To verify that it fails on the old application code:

```
git checkout master
git checkout nanotone/itf-ep-code-bug spec/
bundle exec rspec spec/models/claim_review_spec.rb
```

### CODE REVIEW

**Review by Sjones352 — APPROVED** [review]
LGTM

**Comment by nanotone** [comment]
The much-simpler approach doesn't work because it saves the RequestIssue prematurely (with nil benefit_type) and was causing hundreds more test failures 😅

**Comment by nanotone** [comment]
I went back and forth between the two approaches a few times, because it was hard to decide which way would be easier for fixing tests. I think I'm happy with the simpler approach, and this PR should be ready for re-review @Sjones352.

File: app/models/claim_review.rb [hunk 1]
```diff
@@ -85,7 +85,8 @@ def finalized_decision_issues_before_receipt_date
   # Save issues and assign it the appropriate end product establishment.
   # Create that end product establishment if it doesn't exist.
   def create_issues!(new_issues, request_issues_update = nil)
-    new_issues.each do |issue|
+    # Persist all issues to DB first, with a non-null benefit type
+    new_issues.each { |ri| ri.update(benefit_type: "") }.each do |issue|
```
Comment (Sjones352) [inline app/models/claim_review.rb:89]: @nanotone can you please explain this method? Why are we updating the benefit_type?

**Review by Sjones352 — APPROVED** [review]
Tested it.... LGTM

File: app/models/claim_review.rb (hunk 1 above)
Comment (nanotone) [reply app/models/claim_review.rb:89]: The bugfix requires doing an initial phase of saving all the RequestIssues to the DB first, before a second phase of assigning them to EPEs. Normally I would just do `ri.save!` here, but at this point in time they are missing their benefit_type and the DB requires benefit_type not to be null.

So I'm setting benefit_type to an empty string just so they can be saved to the DB, and the very next line will assign the proper benefit_type.

**Pull request merged**